File: CabService/src/main/java/co/inteleap/course/Ride.java
package co.inteleap.course;

import java.util.function.Supplier;

public class Ride {

    private double Distance;
    private double Time;
    public static final int FARE_PER_KM = 10;
    public static final int FARE_PER_MINUTE = 1;

    public Ride(double distance, double time) {
        this.Distance = distance;
        this.Time = time;
    }

    public Supplier<Double> getTotalFare =
            () -> FARE_PER_KM * Distance + FARE_PER_MINUTE * Time;


}

File: TDD-Kotlin/src/main/kotlin/Ride.kt
data class Ride(val time: Double, val distance: Double) {

    val FARE_PER_KM = 10
    val FARE_PER_MINUTE = 1

    fun Fare(): Double {
        return time * FARE_PER_MINUTE + distance * FARE_PER_KM
    }


}
